### DISCUSSION TITLE
Error migrating code from aws code commit to github

### DISCUSSION THREAD

**Discussion by tthoudam** [community/community#131352] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am trying to migrate code from aws code commit to github. The migration is failing with the following error:
`
"error: RPC failed; HTTP 500 curl 22 The requested URL returned error: 500
fatal: the remote end hung up unexpectedly"
`

I believe the reason is because git fsck is failing becaues of a bad commit in the history.
`"error in commit xxxx: badName: invalid author/committer line - bad name"`

I can see that some commits have an invalid email.
What is the best way to fix this? Is there a way to disable fsck check in github?

**Comment by mustafaobaidd** [comment]
The specific error message "error: RPC failed; HTTP 500 curl 22 The requested URL returned error: 500 fatal: the remote end hung up unexpectedly" indicates that there are problems with the commits in the repository.

The error "error in commit xxxx: badName: invalid author/committer line - bad name" suggests that some commits have invalid email addresses associated with them.

To address this issue, you can try the following steps:

1. Identify and Fix Invalid Commits:
   - You can use Git tools to identify the specific commits that have invalid author/committer lines with bad names or email addresses. Once identified, you can amend these commits to correct the information.
  
2. Disable fsck Check:
   Unfortunately, GitHub does not provide a direct option to disable the fsck check on the server side. However, you can disable the fsck check locally in your Git configuration by setting the "core.fsckObjects" configuration option to false. This will prevent Git from running the fsck check during operations.

Here is how you can disable fsck check locally:
```
git config core.fsckObjects false
```

After disabling the fsck check locally, you can attempt the migration to GitHub again. Make sure to fix any invalid commits before migrating the code.

Please note that it is important to ensure that all commits in the repository are valid and correct before pushing them to GitHub to maintain the integrity of your codebase.
